Add The Tiberium Strain (CnC, scb31ea)

JovialFeline opened the pull request:
This adds a bite-sized mission for Nod from Covert Ops. It is a bit easier than its original form, partly because friendly fire is less brutal.

## Known issues
- Like other TD maps, narrow chokepoints can cause some inconvenient pathing. The canyon after the second bridge may feel extra awkward.
  - There's not much to do on the map script's side, but most vehicles that die on a bridge will not create husks.
- ~~At the moment, armed civilians do not properly Hunt because they lack autotargeting. This is an issue shared with `nod4b`.~~ (https://github.com/OpenRA/OpenRA/pull/21781)

## Notable differences from original
### GDI base and structures
- Change the owner of the bio centers to GDI (from Neutral).
- Add flare smoke and minor vision to destroyed bio centers.
- Add a warning reveal for the towers at the west entrance of the GDI base.
- Change the south gate's footprint to cover the possibility of destroyed concrete walls.
- Change the lone Grenadier by the bio centers to an area guard.
- Skip the southeast Communications Center potentially calling for help from idle GDI soldiers.
  - To be clear, this had nothing to do with the mission script, but rather the original game's AI.

### Other
- Add extra reinforcements for the player if they're on Easy difficulty, and more enemies for Hard.
- Change village houses to require force-fire if on Easy difficulty.
- Add some Nod chit-chat about the villager objective.
- Add a more dramatic entrance to the Flame Tank that appears after the village tank's death.
- Skip the village tank's area guard routine so it's less likely to charge ahead of its rifle escort.
- Add a Nod clean-up crew that arrives in the main village if civilians still linger there. This only happens once all of GDI's units and bio centers are destroyed.

## TD '95 behavior and conversion quirks
- Because the bio centers were owned by Neutral, the AI could order villagers to defend "their" bio centers. Only the armed ones were used (`c1` and `c7`) and they could defend houses as well. The first behavior is skipped, but the main village houses can summon ~~mild distractions~~ help when damaged.
- The bio centers were less than sturdy, comparable to an Advanced Power Plant. They could be brought down by 10 shots from a Flame Tank or 12 from a Chemical Warrior. The default skirmish rules in ORA make it stronger at 34/90 shots, so it is adjusted to be less tedious.
- It seemed intended that all GDI units were wiped out for a Nod victory. However, the `gdix` trigger was incorrectly set and only civilian deaths were necessary.
  - The same behavior applies to the converted mission, at least for now. I don't think it rubs against the briefing too much.
- Reinforced Humvees from the east attack-moved toward the west gate and then hunted. In this version, they move toward the south gate before attempting to move inside. This makes them less likely to circle all the way around because of a blockage. The original game's pathing was more stubborn about cases like this.
- The helicopter that lands on the southern road came from the east edge. Because Flame Tanks in ORA are _much_ faster than their '95 counterparts, the helicopter now starts much closer. Like the eastern Humvees, the passengers use added waypoints to avoid circling.
- Rifle infantry from '95 had a little less range. The ORA start positions for Nod are shifted north by one cell to avoid getting shot before the player does anything.

Porenutak commented:
I didn't play the original mission, so I dont know if these suggestions are out of place or not:

- I suggest make village buildings forcefire only (at least on easy difficluty). I often miss click on building instead of civilian right next to it and lose.
- On hard difficluty add more solo rockets squads instead of bigger existing squads. So you are forced to use more infantry, not just tanks.

Otherwise looks good.

JovialFeline commented:
Rebased and tweaked.

- Fixed GuardArea's premature activation.
- Changed the bridge husk skip to include Hum-vees.
- Easy: Changed village structures to require force-fire to target.
- Hard: Added one Medium Tank each to the west village entrance and the southern river crossing.
- Hard: Added bazooka soldiers near the northern bridge.

PunkPun commented on mods/cnc/fluent/lua.ftl:
```diff
@@ -141,14 +141,22 @@ destroy-capture-warfactory = Destroy or capture the Weapons Factory.
+destroy-gdi-bio-centers = Destroy the GDI bio centers.
+keep-all-other-structures = Keep all other structures intact.
+kill-every-villager-in-area = Kill every villager in the area.
+careless-gdi-experiments-killed = Careless GDI experiments killed these villagers. They must not live to dispute that story.
```
I recon this sentense is a bit too long. Maybe split it into two notifications?

PunkPun commented on mods/cnc/maps/the-tiberium-strain/rules.yaml:
```diff
@@ -0,0 +1,90 @@
+	LuaScript:
+		Scripts: campaign.lua, utils.lua, the-tiberium-strain.lua
+	MissionData:
+		Briefing: A nearby GDI base is conducting further ion research. In doing so, they use a large number of chemicals, located in their bio centers. Destroy all the bio centers, and "contaminate" all units and civilians. Leave all the other structures intact, so that it looks like an accident.
```
could this be extracted into fluent? Run the --extract-yaml-strings command

PunkPun reviewed (commented):
Man the mission is brutal on hard. It's still beatable.

Left a comment, otherwise LGTM. I only glanced over lua

JovialFeline commented:
Rebased, separated the two Nod sentences into their own messages, and added a check to skip those messages if Nod is wiped out.

JovialFeline committed:
Add The Tiberium Strain mission (CnC)

JovialFeline commented:
Added small difficulty changes for Hard and a tweak for the sweepers.

- The extra Medium Tanks avoid giving chase; Flame Tanks will find it easier to disengage.
- The base's south gate is revealed if the Mammoth Tank moves there.
- Nod's west village sweepers can reinforce if the Mammoth Tank is the only remaining GDI unit.
- While sweeper reinforcements are pending, Nod deaths can not trigger defeat.

PunkPun reviewed (approved):
Feels great

PunkPun commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/e33f0f06095cc60c6a505bacbe44bff2ebb30e98...fa6c15783f14d5287f0ff6299de8bb2189bf3ddb
